local/e2e/compose/compose_test.go

```diff
@@ -192,15 +192,19 @@ func TestDownComposefileInParentFolder(t *testing.T) {
 func TestAttachRestart(t *testing.T) {
 	c := NewParallelE2eCLI(t, binDir)
 
-	res := c.RunDockerOrExitError("compose", "--ansi=never", "--project-directory", "./fixtures/attach-restart", "up")
+	cmd := c.NewDockerCmd("compose", "--ansi=never", "--project-directory", "./fixtures/attach-restart", "up")
+	res := icmd.StartCmd(cmd)
 	defer c.RunDockerOrExitError("compose", "-p", "attach-restart", "down")
-	output := res.Stdout()
 
-	exitRegex := regexp.MustCompile("another_1 exited with code 1")
-	assert.Equal(t, len(exitRegex.FindAllStringIndex(output, -1)), 3, res.Combined())
+	c.WaitForCondition(func() (bool, string) {
+		out := res.Stdout()
+		debug := res.Combined()
+		exitRegex := regexp.MustCompile("another_1 exited with code 1")
+		return len(exitRegex.FindAllStringIndex(out, -1)) == 3, fmt.Sprintf("'another_1 exited with code 1' not found 3 times in : \n%s\n", debug)
```

Comment:
Do we need a regex for this? Can’t we use [strings.Count](https://golang.org/pkg/strings/#Count)?

Reply:
indeed, not sure why we had a regex initially there... fixed